addons/pos_six/static/src/js/payment_six.js

```diff
@@ -28,6 +28,19 @@ var PaymentSix = PaymentInterface.extend({
         this._super.apply(this, arguments);
         this.enable_reversals();
 
+        var terminal_ip = arguments[1].six_terminal_ip
+        var instanced_payment_methods = _.filter(this.pos.payment_methods, function(payment_method) {
```

Comment:
As you only need one element, you can use "_.find" instead of filter, you'll also have to change the if to check if !== undefined and instanced_payment_methods[0] by instanced_payment_methods (you can also remove the "s" in variable name as there will be only one object)